hibernate-core/src/main/java/org/hibernate/dialect/MariaDB103Dialect.java

```diff
@@ -57,7 +57,7 @@ public String getSelectSequenceNextValString(String sequenceName) {
 
 	@Override
 	public String getQuerySequencesString() {
-		return "select table_name from information_schema.TABLES where table_type='SEQUENCE'";
+		return "select table_name from information_schema.TABLES where table_schema = database() and table_type = 'SEQUENCE'";
```

Comment:
@NathanQingyangXu so is this syntax really specific to MariaDB 10.3, or is it also correct on earlier versions?

Reply:
I tried to get the answer from https://mariadb.com/kb/en/database/ to no avail. But given the page was created 9 years ago, it seems to have a long history.

Reply:
I checked the MySQL info for v5.5 and found this:https://dev.mysql.com/doc/refman/5.5/en/information-functions.html#function_database.
So the function exists from the beginning of Mariadb.

Reply:
Would you mind my asking why you asked the question?x

Reply:
If your concern is more about sequence, then yeah, it is only in this version that mariadb supports sequence, because of some Chinese engineers open source contribution, to my knowledge.

Reply:
OK thanks that’s what I needed to know (I need to replicate this change on the Hibernate 6 branch).

Reply:
OK thanks that’s what I needed to know (I need to replicate this change on the Hibernate 6 branch).